contrib_bots/lib/github.py

```diff
@@ -0,0 +1,63 @@
+# The purpose of this file is to handle all requests
+# to the Github API, which will make sure that all
+# requests are to the same account, and that all requests
+# authenticated correctly and securely
+from __future__ import absolute_import
+from future import standard_library
+standard_library.install_aliases()
+import json
+import os
+import requests
+import six.moves.configparser
+import urllib.request, urllib.error, urllib.parse
+
+class github(object):
+    '''
+    Before use, check ~/.github/github.conf, and make sure there are
+    github_repo   (The name of the repo to post to)
+    github_repo_owner   (The owner of the repo to post to)
+    github_username   (The username of the github bot)
+    github_token    (The personal access token for the github bot)
+    '''
+    
+    CONFIG_FILE = '~/.github/github.conf'
+    BASE_URL = ''
+    SESSION = requests.session()
+    AUTHED = False
+
+    def auth(self):
+        # Sets up the configParser to read the .conf file
+        config = six.moves.configparser.ConfigParser()
+        # Reads the config file
+        config.read([os.path.expanduser(self.CONFIG_FILE)])
+        # Gets the properties from the config file
+        oauth_token = config.get('github', 'github_token')
+        username = config.get('github', 'github_username')
+        self.BASE_URL = 'https://api.github.com/repos/{}/{}'.format(config.get('github', 'github_repo_owner'), config.get('github', 'github_repo'))
```

Comment:
`self.BASE_URL` will probably work but just to clarify you are not updating `github.BASE_URL` variable you defined in line 24 so make sure that there are no unwanted side effects.